File: Client/frontend/src/app/models/leaves.ts
import {LeaveType} from "./leaveType";
import {LeaveStatus} from "./Enum/LeaveStatus";

export interface ILeave {
  id?:number,
  startDate?:Date,
  endDate?:Date,
  leaveReason?:string,
  employeeId?:number,
  RemainingLeave?:number,
   numberOfDays?:number,
  leaveType?: LeaveType,
  status?:LeaveStatus
}

export class Leave implements ILeave {
  constructor(public id:number,
              public startDate?:Date,
              public endDate?:Date,
              public leaveReason?:string,
              public employeeId?:number,
             public RemainingLeave?:number,
             public numberOfDays?:number,
              public leaveType?: LeaveType,
              public status?:LeaveStatus
              ) {
  }
}

File: Configuration/src/main/java/com/medecineWebApp/Configuration/models/Leaves.java
package com.medecineWebApp.Configuration.models;

import com.medecineWebApp.Configuration.enums.LeaveStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.io.Serializable;
import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "Leaves")
@EntityListeners(AuditingEntityListener.class)
public class Leaves extends Auditable implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDate startDate;

    private LocalDate endDate;

    private String leaveReason;
    private Long RemainingLeave;

    private Long employeeId;


    @ManyToOne
    @JoinColumn(name = "leave_type_id")
    private LeaveType leaveType;
    private LeaveStatus status;

}
